7_react/react1/week3/public/src/components/List.js

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect, useState } from "react";
+import ListItem from "./ListItem";
+import ReactDatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+
+
+export function List() {
+  const [list, setList] = useState(initialList);
+  const [newTodo, setNewTodo] = useState('');
+  const [newDeadline, setNewDeadline] = useState('');
+
+
+  useEffect(() => {
+
+    fetch('https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw')
+      .then(response => response.json())
+      .then(data => setList(data))
+      .catch(error => console.error('Error fetching todos:', error));
+  }, []);
+
+  const addRandomTask = () => {
+    
+    const newTask = {
+      id: Date.now(),
+      description: newTodo,
+      deadline: newDeadline,
+      done: false,
+    };
+
+    setList((previousState) => [...previousState, newTask]);
+    setNewTodo('');
+      setNewDeadline('');
```

Comment:
I would recommend getting into the habit of formatting your code before committing, so the code looks neater.
You can even set VS code to to this automatically every time you save or after every few seconds.
See https://code.visualstudio.com/docs/editor/codebasics#_formatting